### ISSUE TITLE
ad_modify_password has inconsistent response when not raising exceptions

### ISSUE THREAD

**Issue by ashleybartlett** [cannatag/ldap3#371]
This makes it hard to do simple tests on. I now need to check type that is returned when checking if a modify_password request is successful.

Should just return the result always. doing 
```python
r = conn.extend.microsoft.modify_password(user, password)
if r:
    print("This will always be called")
```

I can't plainly do, and I shouldn't have to enable raise exceptions because one function has a different return type.
``` python
if not result['result'] == RESULT_SUCCESS:
    # Do something about this. Or get exception, because result is True.
```

**Comment by cannatag** [comment]
Please check code in dev, now ad_modify_password returns False when change is not successful

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.
